test/tests/functional/pbs_job_task.py

```diff
@@ -117,15 +117,17 @@ def test_invoke_pbs_tmrsh_from_sister_mom(self):
         mom1 = self.moms.keys()[0]
         mom2 = self.moms.keys()[1]
         mom3 = self.moms.keys()[2]
+        fqdn_mom2 = socket.getfqdn(mom2)
+        fqdn_mom3 = socket.getfqdn(mom3)
```

Comment:
what if getfqdn fails?

Reply:
why would getfqdn fail? if we run with wrong hostname values to -p argument? 
PTL would catch that in setup() before starting the test.

Reply:
PTL setUp() doesn't use getfqdn(), see PBSService.\_\_init__() (https://github.com/openpbs/openpbs/blob/master/test/fw/ptl/lib/ptl_service.py#L509)

So, you can use self.moms.values()[0].hostname (or .fqdn) to get FQDN of that mom

Reply:
Thanks @hirenvadalia. I would still like to understand why getfqdn() could cause a problem

Reply:
@saimanikant 
I didn't say getfqdn() will cause a problem, my point is whatever you are trying to do is already there, so use it.

@prakashcv13
getfqdn will never fail, it will try to find fqdn if available, if not then it will return hostname returned by socket.gethostname() (which can be fqdn or short name, based on system configuration)

Reply:
@hirenvadalia and @saimanikant - I should have been clear - by 'fail' i mean what if getfqdn() returns the short name itself. what should the test case do in that case.

Reply:
@prakashcv13, My understanding is that pbs_tmrsh rejects with usage of short name and we expect pbs_tmrsh to work with hostnames listed in $PBS_NODEFILE and $PBS_NODEFILE shows Fqdn names. 
I don't think the fqdn names in $PBS_NODEFILE will be different from the values returned by getfqdn()?

Reply:
I am not of the view that we can be absolutely sure about both being same, I would recommend skipping the test case or failing purposely if getfqdn returns the short name. what do you think?

Reply:
I don't think that is necessary. I think the value returned by getfqdn() would be the same value written in $PBS_NODEFILE by PBS.

Reply:
alright. do we have logs supporting your statement. If not, let us err on the side of caution and add the skip/fail code.

Reply:
The logs for the test run with changes are attached in PR description, where pbs_tmrsh runs as expected.

Reply:
well, I am not sure how can I be more clear about this. I would go out on a limb and say that we add the code to check if getfqdn() returns a value equal to the short name and skip/fail the test if needed.